Staging next

jonringer opened the pull request:
https://hydra.nixos.org/job/nixpkgs/staging-next/unstable#tabs-constituents (soon)

These changes will not be included in 21.05 unless specifically backported. Staging-next has returned to a normal workflow.

Previous Staging-next: #123279

primeos committed:
mesa: 21.0.3 -> 21.1.1

Note: This update likely causes some issues when running an application
that has a direct dependency on Mesa (e.g. Sway and XWayland) and was
compiled against a different Nixpkgs revision. See 7106fca0fe4 for more
details regarding that issue.

primeos committed:
mesa: 21.0.1 -> 21.0.3

Note: The update to Mesa 21.0.2 was reverted (25ae1fd29f8) because it
caused major issues with Sway (segfault on startup [0]).
This is still the case and might affect all packages that directly
depend on "mesa" (for libgbm or libglapi) but it only causes issues when
the package depends on a "mesa" version that differs from "mesa.drivers"
used for "/run/opengl-driver/". I've noticed this while testing Mesa
updates with the NixOS option "hardware.opengl.package" (as usual)
instead of rebuilding my whole system (which would work). Unfortunately
this can/will likely also cause issues when mixing different channels,
using Flakes/Overlays, etc.

The cause of this should be similar to [1] ("mesa" updates now cause the
same issues that "glibc" updates already do, maybe triggered by certain
Mesa changes) and some additional discussions is in [2],[3].

Note: Don't backport this to NixOS 21.05, at least not without careful
consideration.

[0]: https://github.com/NixOS/nixpkgs/pull/118753#issuecomment-818950977
[1]: https://github.com/NixOS/nixpkgs/issues/95808
[2]: https://github.com/NixOS/nixpkgs/pull/120325
[3]: https://github.com/NixOS/nixpkgs/pull/119558

primeos committed:
mesa: Cleanup enableRadv (not used anymore)

r-ryantm committed:
dbus-glib: 0.110 -> 0.112

jtojnar committed:
dbus-glib: update license

r-ryantm committed:
libgcrypt: 1.9.2 -> 1.9.3

r-ryantm committed:
libksba: 1.5.0 -> 1.5.1

r-ryantm committed:
libjpeg: 2.0.6 -> 2.1.0

primeos committed:
wayland-protocols: Switch to Meson

According to diffoscope this is fine (only the Nix store path changes).

Cross-compilation with doCheck=true is broken (requires pkg-config in
nativeBuildInputs but doesn't play nicely with checkInputs).

Commit:
tcl: Add tclPackageHook and mkTclDerivation

Implement tclPackageHook, a setup hook that adjusts TCLLIBPATH to include the
paths of any installed Tcl packages, propagates that TCLLIBPATH to
anti-dependencies, and wraps any installed binaries to set their TCLLIBPATH.

Additionally, implement a makePythonPackage-style mkDerivation wrapper to use
reasonable defaults for Tcl packages and use tclPackageHook.

ajs124 committed:
libgcrypt: add meta.changelog

Commit:
tclreadline: simplify package with tcl.mkTclDerivation

Commit:
incrtcl: Simplify package with tcl.mkTclDerivation

Commit:
itktcl: Simplify package with tcl.mkTclDerivation

Commit:
tcllib: Simplify package with tcl.mkTclDerivation

Commit:
tclx: Leverage tcl.mkTclDerivation

Commit:
tix: Simplify package with tcl.mkTclDerivation

Commit:
tcltls: Simplify package with tcl.mkTclDerivation

Commit:
tk: Simplify package with tcl.mkTclDerivation

Commit:
expect: Simplify package with tcl.mkTclDerivation

Commit:
poke: Simplify package with tcl.tclPackageHook

Commit:
remind: Simplify package with tcl.mkTclDerivation

Commit:
fossil: Remove unneeded TCLLIBPATH export

The use of tcl.mkTclDerivation in tcllib ensures that it will
already be present on TCLLIBPATH when tests are run.

Commit:
bwidget: Simplify bwidget package with tcl.mkTclDerivation

Commit:
tcl2048: Simplify package with tcl.mkTclDerivation

Commit:
xconq: Simplify package with tcl.tclPackageHook

Commit:
maintainers: add agbrooks

Commit:
scid: Simplify package with tcl.mkTclDerivation

Commit:
scid-vs-pc: Simplify package with tcl.mkTclDerivation

Commit:
tix: Prevent tcl.mkTclDerivation from altering configureFlags

mweinelt committed:
python3Packages.sqlalchemy: 1.3.23 -> 1.4.15

mweinelt committed:
python3Packages.marshmallow-sqlalchemy: 0.24.2 -> 0.25.0

mweinelt committed:
python3Packages.marshmallow: 3.10.0 -> 3.11.1

markuskowa committed:
openblas: 0.3.13 -> 0.3.15

agbrooks committed:
scid: Correct typo in maintainers

Commit:
Revert "xconq: Simplify package with tcl.tclPackageHook"

This reverts commit 113826a892512eb2e19985310fd7d4a39db4d75e, which
ended up breaking the Tk import at runtime.

Commit:
bwidget: add missing dependency on tk

fabaff committed:
python3Packages.importlib-metadata: 3.7.3 -> 4.0.1

Commit:
tcl: fix TCLLIBPATH edge cases in tcl package hook

TCLLIBPATH was not actually extended if a package's path was a substring
of a package already on the path. This commit fixes that behavior.

Additionally, '--prefix' is now used in place of '--set', as the latter
was found to break tkremind from the remind package.

r-ryantm committed:
chibi: 0.9.1 -> 0.10

agbrooks committed:
poke: revise formatting

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

agbrooks committed:
tcl: combine inherit expressions

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

agbrooks committed:
bwidget: revise formatting

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

agbrooks committed:
bwidget: make installPhase slightly clearer

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

agbrooks committed:
bwidget: replace verbose let/in expression with rec

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

agbrooks committed:
tclx: simplify postInstall phase

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

Commit:
tcl: avoid readonly local in package hook

If a path with a space were added to TCLLIBPATH, _addToTclLibPath
would attempt to mutate the tclPkg local and fail.

Commit:
tcl: use double square brackets consistently in tcl package hook

Commit:
tclx: fix undefined variable error

lblasc committed:
rustc: 1_52, use correct llvm version

bachp committed:
openethereum: 3.2.5 -> 3.2.6

Override of LLVM is no longer requires as of rust 1.52

erictapen committed:
imagemagick: 7.0.11-9 -> 7.0.11-12

primeos committed:
dav1d: 0.8.2 -> 0.9.0

r-ryantm committed:
jmol: 14.31.38 -> 14.31.39

r-ryantm committed:
libqmi: 1.28.2 -> 1.28.4

github-actions[bot] committed:
Merge staging-next into staging

r-ryantm committed:
just: 0.9.1 -> 0.9.3

r-ryantm committed:
k9s: 0.24.8 -> 0.24.9

eduardosm committed:
pythonPackages.buildPythonPackage: enable verbose output with `pyproject`

This makes the build log more useful

primeos committed:
cargo-c: 0.8.0 -> 0.8.1

primeos committed:
Merge pull request #123268 from primeos/dav1d

dav1d: 0.8.2 -> 0.9.0

primeos committed:
Merge pull request #121727 from primeos/wayland-protocols-switch-to-meson

wayland-protocols: Switch to Meson

sternenseemann committed:
pkg-config-wrapper: mangle PKG_CONFIG_PATH{,_FOR_BUILD} correctly

Previously, mangleVarList would be used which would concatenate the
variables using a space as a separator. Paths are however separated by
`:` in PKG_CONFIG_PATH leading to entries being broken.

This is fixed by introducing mangleVarListGeneric which allows us to
specify the desired separator.

Reproducer for the issue prior to this change:

    $ nix-shell -A pkgsLLVM.wayland
    [nix-shell] $ pkg-config --libs expat
    Package expat was not found in the pkg-config search path.
    Perhaps you should add the directory containing `expat.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'expat' found
    $ printf 'Host: %s\nBuild: %s' $PKG_CONFIG_PATH $PKG_CONFIG_PATH_FOR_BUILD
    Host: /nix/store/5h308a4ab8w7prcp8iflh5pnl78mayi2-expat-2.2.10-x86_64-unknown-linux-gnu-dev/lib/pkgconfig:/nix/store/z3y9ska2h4l1map25m195iq577g7g3gz-libxml2-x86_64-unknown-linux-gnu-2.9.12-dev/lib/pkgconfig:/nix/store/lbz5m1s0r7zn0cxvl21czfspli6ribzb-zlib-1.2.11-x86_64-unknown-linux-gnu-dev/lib/pkgconfig:/nix/store/rfhvp8r8n3ygpzh8j0l34lk8hwwi3z0h-libffi-3.3-x86_64-unknown-linux-gnu-dev/lib/pkgconfig
    Build: /nix/store/dw11ywy7qwfz53qisz0dggbgix88jah2-wayland-1.19.0-bin/lib/pkgconfig

strace reveals the issue:

    stat("/nix/store/dw11ywy7qwfz53qisz0dggbgix88jah2-wayland-1.19.0-bin/lib/pkgconfig /nix/store/5h308a4ab8w7prcp8iflh5pnl78mayi2-expat-2.2.10-x86_64-unknown-linux-gnu-dev/lib/pkgconfig/expat-uninstalled.pc", 0x7fff49829fa0) = -1 ENOENT (No such file or directory)

In the pkg-config wrapper $PKG_CONFIG_PATH_FOR_BUILD and
$PKG_CONFIG_PATH are concatenated with a space which leads to two paths
being messed up. This issue likely only affects native cross
compilation.

Ericson2314 committed:
Merge pull request #123415 from sternenseemann/fixed-pkg-config-wrapper-mangling

pkg-config-wrapper: mangle PKG_CONFIG_PATH{,_FOR_BUILD} correctly

Atemu committed:
libndctl: build docs with asciidoc instead of asciidoctor

Removes dependency on ruby 2.7 and isn't much slower

primeos committed:
Merge pull request #123371 from primeos/cargo-c

cargo-c: 0.8.0 -> 0.8.1

primeos committed:
libdrm: 2.4.105 -> 2.4.106

sternenseemann committed:
llvmPackages*.clang: fix linker invocation with LLVMgold plugin

When using GNU binutils, clang passes the LLVMgold.so plugin to the
linker for certain operations that require special support in the linker
like doing link time optimization (LTO). When passing the plugin to the
linker's command line, clang assumes that llvm and itself are installed
in the same prefix and thus `/path/to/clang/bin/../lib/LLVMgold.so` is
the plugin.

Since we install clang and llvm to separate store paths, this assumption
does not hold. When clang-unwrapped only had a single output, we worked
around this issue by symlinking `$out/lib/LLVMgold.so` to
`${llvm}/lib/LLVMgold.so`. However since we split all llvm packages into
multiple outputs clang's `$out` no longer has a lib directory and clang
can't discover clangs lib output on its own. As a result LTO was broken.

Instead of introducing yet another hack and having a symlink to
LLVMgold.so in `$out/lib` (despite having `$lib/lib` as well), we patch
clang to use a hard coded path to `${libllvm.lib}/lib` for discovering
`LLVMgold.so`.

Resolves #123361.

ttuegel committed:
Merge pull request #122035 from markuskowa/upd-openblas

openblas: 0.3.13 -> 0.3.15

github-actions[bot] committed:
Merge staging-next into staging

dotlambda committed:
Merge pull request #123225 from erictapen/imagemagick-7.0.11-12

imagemagick: 7.0.11-9 -> 7.0.11-12

r-ryantm committed:
kodiPackages.inputstream-adaptive: 2.6.14 -> 2.6.15

Ericson2314 committed:
Merge pull request #123551 from sternenseemann/clang-fix-llvmgold-path

llvmPackages*.clang: fix linker invocation with LLVMgold plugin

github-actions[bot] committed:
Merge staging-next into staging

aanderse committed:
Merge pull request #123577 from r-ryantm/auto-update/kodi-inputstream-adaptive

kodiPackages.inputstream-adaptive: 2.6.14 -> 2.6.15

TredwellGit committed:
xorg.libX11: 1.7.0 -> 1.7.1

https://lists.x.org/archives/xorg-announce/2021-May/003089.html
https://lists.x.org/archives/xorg-announce/2021-May/003088.html

Atemu committed:
Revert "git: Use asciidoctor instead of asciidoc for manpages"

This reverts commit bbf96d898b96ea2d1ff19a09e545499e4f32d8d0.

Removes dependency on ruby 2.7 and isn't much slower.

See https://github.com/NixOS/nixpkgs/pull/123502 for more info

github-actions[bot] committed:
Merge staging-next into staging

Mindavi committed:
libwacom: 1.9 -> 1.10

jtojnar committed:
Merge pull request #123733 from Mindavi/libwacom/1.10

libwacom: 1.9 -> 1.10

baloo committed:
rubygems: ensure temporary directories are cleaned up

see #123718

Signed-off-by: Arthur Gautier <baloo@superbaloo.net>

marsam committed:
libsass: 3.6.4 -> 3.6.5

https://github.com/sass/libsass/releases/tag/3.6.5

primeos committed:
Merge pull request #123519 from primeos/libdrm

libdrm: 2.4.105 -> 2.4.106

github-actions[bot] committed:
Merge staging-next into staging

primeos committed:
Merge pull request #120325 from primeos/mesa

mesa: 21.0.1 -> 21.0.3

github-actions[bot] committed:
Merge staging-next into staging

marsam committed:
Merge pull request #123748 from baloo/baloo/rubygems/tmpdir

rubygems: ensure temporary directories are cleaned up

vcunat committed:
Merge #123622: xorg.libX11: 1.7.0 -> 1.7.1

jtojnar committed:
libxml2: Work around lxml API misuse

github-actions[bot] committed:
Merge staging-next into staging

raboof committed:
Merge pull request #123502 from Atemu/minimalIso-remove-ruby2.7

minimalIso: remove ruby2.7

github-actions[bot] committed:
Merge staging-next into staging

Commit:
tcl: update maintainers

When I reached out to vrthra / Rahul Gopinath about this PR, he
indicated that he no longer has the time to serve as tcl maintainer
and suggested that I take on the responsibility.

7c6f434c committed:
Merge pull request #123281 from r-ryantm/auto-update/jmol

jmol: 14.31.38 -> 14.31.39

Mindavi committed:
atk: support cross-compilation by disabling introspection

7c6f434c committed:
Merge pull request #122472 from agbrooks/tcl-packaging

tcl: add Tcl packaging support

Commit:
tcltls: 1.6.7 -> 1.7.21

Upgrade tcltls and use new tcl-lang.org homepage.
This version no longer needs to be built against openssl_1_0_2.

Commit:
tcltls: remove unused stdenv argument

github-actions[bot] committed:
Merge staging-next into staging

marsam committed:
Merge pull request #123853 from marsam/update-libsass

libsass: 3.6.4 -> 3.6.5

github-actions[bot] committed:
Merge staging-next into staging

7c6f434c committed:
Merge pull request #123923 from Mindavi/atk/cross-compile

atk: support cross-compilation by disabling introspection

r-ryantm committed:
btrfs-progs: 5.11.1 -> 5.12.1

github-actions[bot] committed:
Merge staging-next into staging

7c6f434c committed:
Merge pull request #124007 from r-ryantm/auto-update/btrfs-progs

btrfs-progs: 5.11.1 -> 5.12.1

jonringer committed:
Merge remote-tracking branch 'origin/staging' into staging-next

r-rmcgibbo commented:
Result of `nixpkgs-review pr 124095` at b9b9cdb8 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1528 packages marked as broken and skipped:</summary>
  <ul>
    <li>SDL_sixel</li>
    <li>adobe-reader</li>
    <li>adoptopenjdk-hotspot-bin-13</li>
    <li>adoptopenjdk-hotspot-bin-14</li>
    <li>adoptopenjdk-jre-hotspot-bin-13</li>
    <li>adoptopenjdk-jre-hotspot-bin-14</li>
    <li>adoptopenjdk-jre-openj9-bin-13</li>
    <li>adoptopenjdk-jre-openj9-bin-14</li>
    <li>adoptopenjdk-openj9-bin-13</li>
    <li>adoptopenjdk-openj9-bin-14</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>35437 packages skipped due to time constraints:</summary>
  <ul>
    <li>AMB-plugins</li>
    <li>AusweisApp2</li>
    <li>CHOWTapeModel</li>
    <li>CoinMP</li>
    <li>DisnixWebService</li>
    <li>EBTKS</li>
    <li>EmptyEpsilon</li>
    <li>FIL-plugins</li>
    <li>Fabric (python38Packages.Fabric)</li>
    <li>LAStools</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>622 packages built successfully:</summary>
  <ul>
    <li>alegreya-sans</li>
    <li>amoeba-data</li>
    <li>asdf</li>
    <li>atk (gnome.atk ,gnome2.atk)</li>
    <li>bash (runtimeShellPackage)</li>
    <li>bfetch</li>
    <li>bintoolsNoLibc (binutilsNoLibc)</li>
    <li>bubblewrap</li>
    <li>bwidget</li>
    <li>ccloud-cli</li>
    <li>chibi</li>
    <li>commonsFileUpload</li>
    <li>copyDesktopItems</li>
    <li>copyright-update</li>
    <li>dav1d</li>
    <li>dbus-glib</li>
    <li>deer</li>
    <li>dehydrated</li>
    <li>discocss</li>
    <li>disfetch</li>
    <li>docbook-xsl-ns (docbook_xsl_ns)</li>
    <li>docbook_sgml_dtd_41</li>
    <li>dwmblocks</li>
    <li>edusong</li>
    <li>eigen</li>
    <li>eject (unixtools.eject)</li>
    <li>enableGCOVInstrumentation</li>
    <li>essentia-extractor</li>
    <li>expect</li>
    <li>fixup_yarn_lock</li>
    <li>fontconfig</li>
    <li>freepats</li>
    <li>gdk-pixbuf</li>
    <li>ghc_filesystem</li>
    <li>gnome.gnome-common (gnome.gnome_common)</li>
    <li>gnomeExtensions.archlinux-updates-indicator</li>
    <li>gnomeExtensions.better-osd-gnome-40</li>
    <li>gnomeExtensions.colortint</li>
    <li>gnomeExtensions.colosseum</li>
    <li>gnomeExtensions.draw-on-your-screen</li>
    <li>gnomeExtensions.espresso</li>
    <li>gnomeExtensions.kimpanel</li>
    <li>gnomeExtensions.mpris-indicator-button</li>
    <li>gnomeExtensions.nasa-apod</li>
    <li>gnomeExtensions.panel-scroll</li>
    <li>gnomeExtensions.right-click-down</li>
    <li>gnomeExtensions.sermon</li>
    <li>gnomeExtensions.worksets</li>
    <li>gnomeExtensions.workspace-switch-wraparound</li>
    <li>gnu-config</li>
    <li>python3 (gnuradioPackages.python ,python38 ,python38Packages.python ,sourcehut.python)</li>
    <li>gsm</li>
    <li>gwtdragdrop</li>
    <li>haskell.compiler.ghc865Binary</li>
    <li>hexdump (unixtools.hexdump)</li>
    <li>hicolor-icon-theme</li>
    <li>hostname (unixtools.hostname)</li>
    <li>hunspellDicts.it-it (hunspellDicts.it_IT)</li>
    <li>ijs</li>
    <li>incrtcl</li>
    <li>installShellFiles</li>
    <li>iodash</li>
    <li>ipcalc</li>
    <li>itktcl</li>
    <li>jiten</li>
    <li>kakounePlugins.connect-kak</li>
    <li>kakounePlugins.kakoune-rainbow</li>
    <li>keepBuildTree</li>
    <li>ladspaH</li>
    <li>ld-is-cc-hook</li>
    <li>libbass_fx</li>
    <li>libdrm</li>
    <li>libevdev</li>
    <li>libimagequant</li>
    <li>libinput</li>
    <li>libjpeg (libjpeg_turbo)</li>
    <li>libksba</li>
    <li>libmicrodns</li>
    <li>libndctl</li>
    <li>libqrtr-glib</li>
    <li>libredirect</li>
    <li>libudev0-shim</li>
    <li>libwacom</li>
    <li>llvmPackages_11.libcxx (llvmPackages_latest.libcxx)</li>
    <li>llvmPackages_11.libcxxabi (llvmPackages_latest.libcxxabi)</li>
    <li>llvmPackages_11.libunwind (llvmPackages_latest.libunwind)</li>
    <li>llvmPackages_5.libcxxabi</li>
    <li>llvmPackages_6.libcxx</li>
    <li>llvmPackages_6.libcxxabi</li>
    <li>lsb-release</li>
    <li>media-player-info</li>
    <li>mkl</li>
    <li>mobile-broadband-provider-info</li>
    <li>monoDLLFixer</li>
    <li>more (unixtools.more)</li>
    <li>mpvScripts.simple-mpv-webui</li>
    <li>mtdev</li>
    <li>multipart-parser-c</li>
    <li>netsurf.buildsystem</li>
    <li>nix-zsh-completions</li>
    <li>npth</li>
    <li>octavePackages.writeRequiredOctavePackagesHook</li>
    <li>opencl-headers</li>
    <li>openvpn_learnaddress</li>
    <li>passExtensions.pass-genphrase</li>
    <li>perl532Packages.AlgorithmDiff</li>
    <li>perl532Packages.BCOW</li>
    <li>perl532Packages.BFlags</li>
    <li>perl532Packages.BHooksEndOfScope</li>
    <li>perl532Packages.CGIEmulatePSGI</li>
    <li>perl532Packages.CanaryStability</li>
    <li>perl532Packages.CarpClan</li>
    <li>perl532Packages.ClassAccessor (perl532Packages.ClassAccessorFast)</li>
    <li>perl532Packages.ClassDataInheritable</li>
    <li>perl532Packages.ClassInspector</li>
    <li>perl532Packages.ClassLoad</li>
    <li>perl532Packages.ClassMethodModifiers</li>
    <li>perl532Packages.ClassTiny</li>
    <li>perl532Packages.Clone</li>
    <li>perl532Packages.CommonSense (perl532Packages.commonsense)</li>
    <li>perl532Packages.CookieBaker</li>
    <li>perl532Packages.CryptPasswdMD5</li>
    <li>perl532Packages.CwdGuard</li>
    <li>perl532Packages.DataDump</li>
    <li>perl532Packages.DataOptList</li>
    <li>perl532Packages.DevelCheckBin</li>
    <li>perl532Packages.DevelCheckCompiler</li>
    <li>perl532Packages.DevelCycle</li>
    <li>perl532Packages.DevelGlobalDestruction</li>
    <li>perl532Packages.DevelOverloadInfo</li>
    <li>perl532Packages.DevelStackTrace</li>
    <li>perl532Packages.DevelStackTraceAsHTML</li>
    <li>perl532Packages.DistCheckConflicts</li>
    <li>perl532Packages.EncodeLocale</li>
    <li>perl532Packages.EvalClosure</li>
    <li>perl532Packages.ExceptionClass</li>
    <li>perl532Packages.ExporterTiny</li>
    <li>perl532Packages.ExtUtilsConfig</li>
    <li>perl532Packages.ExtUtilsHelpers</li>
    <li>perl532Packages.ExtUtilsInstallPaths</li>
    <li>perl532Packages.FileBaseDir</li>
    <li>perl532Packages.FileCopyRecursiveReduced</li>
    <li>perl532Packages.FileShareDir</li>
    <li>perl532Packages.FileShareDirInstall</li>
    <li>perl532Packages.Filepushd</li>
    <li>perl532Packages.HTTPEntityParser</li>
    <li>perl532Packages.HTTPHeadersFast</li>
    <li>perl532Packages.HTTPMultiPartParser</li>
    <li>perl532Packages.HTTPServerSimple</li>
    <li>perl532Packages.HTTPServerSimplePSGI</li>
    <li>perl532Packages.HashMultiValue</li>
    <li>perl532Packages.IPCRun3</li>
    <li>perl532Packages.IPCSystemSimple</li>
    <li>perl532Packages.JSONMaybeXS</li>
    <li>perl532Packages.LWPProtocolhttp10</li>
    <li>perl532Packages.LinguaENWords2Nums</li>
    <li>perl532Packages.LinguaPTStemmer</li>
    <li>perl532Packages.MIMETypes</li>
    <li>perl532Packages.MROCompat</li>
    <li>perl532Packages.MockConfig</li>
    <li>perl532Packages.ModuleBuildTiny</li>
    <li>perl532Packages.ModuleBuildXSUtil</li>
    <li>perl532Packages.ModuleImplementation</li>
    <li>perl532Packages.ModuleManifest</li>
    <li>perl532Packages.ModuleRuntime</li>
    <li>perl532Packages.ModuleRuntimeConflicts</li>
    <li>perl532Packages.MozillaCA</li>
    <li>perl532Packages.NamespaceAutoclean (perl532Packages.namespaceautoclean)</li>
    <li>perl532Packages.NamespaceClean (perl532Packages.namespaceclean)</li>
    <li>perl532Packages.NumberCompare</li>
    <li>perl532Packages.Opcodes</li>
    <li>perl532Packages.POSIXstrftimeCompiler</li>
    <li>perl532Packages.PackageStash</li>
    <li>perl532Packages.ParamsUtil</li>
    <li>perl532Packages.ParseYapp</li>
    <li>perl532Packages.PerlIOutf8_strict</li>
    <li>perl532Packages.Readonly</li>
    <li>perl532Packages.RoleTiny</li>
    <li>perl532Packages.StreamBuffered</li>
    <li>perl532Packages.SubExporter (perl532Packages.SubExporterUtil)</li>
    <li>perl532Packages.SubExporterProgressive</li>
    <li>perl532Packages.SubIdentify</li>
    <li>perl532Packages.SubInstall</li>
    <li>perl532Packages.SubQuote</li>
    <li>perl532Packages.TaskWeaken</li>
    <li>perl532Packages.TestException</li>
    <li>perl532Packages.TestLeakTrace</li>
    <li>perl532Packages.TestNoWarnings</li>
    <li>perl532Packages.TestOutput</li>
    <li>perl532Packages.TestRequires</li>
    <li>perl532Packages.TestTime</li>
    <li>perl532Packages.TestWarn</li>
    <li>perl532Packages.TestWithoutModule</li>
    <li>perl532Packages.TextDiff</li>
    <li>perl532Packages.TextGerman</li>
    <li>perl532Packages.TextGlob</li>
    <li>perl532Packages.TieIxHash</li>
    <li>perl532Packages.TimeDate</li>
    <li>perl532Packages.WWWFormUrlEncoded</li>
    <li>perl532Packages.XMLSAXExpat</li>
    <li>perl532Packages.asa</li>
    <li>pikchr</li>
    <li>polytopes_db</li>
    <li>poppler_data</li>
    <li>ps (unixtools.ps)</li>
    <li>python38Packages.CommonMark</li>
    <li>python38Packages.aniso8601</li>
    <li>python38Packages.astor</li>
    <li>python38Packages.asynctest</li>
    <li>python38Packages.automat</li>
    <li>python38Packages.backcall</li>
    <li>python38Packages.backports-entry-points-selectable</li>
    <li>python38Packages.blinker</li>
    <li>python38Packages.certifi</li>
    <li>python38Packages.constantly</li>
    <li>python38Packages.cookies</li>
    <li>python38Packages.cppy</li>
    <li>python38Packages.cssselect</li>
    <li>python38Packages.cycler</li>
    <li>python38Packages.dateutil (python38Packages.python-dateutil)</li>
    <li>python38Packages.decorator</li>
    <li>python38Packages.defusedxml</li>
    <li>python38Packages.deprecation</li>
    <li>python38Packages.distro</li>
    <li>python38Packages.ecdsa</li>
    <li>python38Packages.entrypoints</li>
    <li>python38Packages.et_xmlfile</li>
    <li>python38Packages.extras</li>
    <li>python38Packages.fixtures</li>
    <li>python38Packages.flask</li>
    <li>python38Packages.future</li>
    <li>python38Packages.greenlet</li>
    <li>python38Packages.hiro</li>
    <li>python38Packages.hyperlink</li>
    <li>python38Packages.ifaddr</li>
    <li>python38Packages.imagesize</li>
    <li>python38Packages.importlib-metadata</li>
    <li>python38Packages.incremental</li>
    <li>python38Packages.ipython_genutils</li>
    <li>python38Packages.isodate</li>
    <li>python38Packages.itsdangerous</li>
    <li>python38Packages.jdcal</li>
    <li>python38Packages.jmespath</li>
    <li>python38Packages.jupyterlab-pygments</li>
    <li>python38Packages.libevdev</li>
    <li>python38Packages.libxml2</li>
    <li>python38Packages.limits</li>
    <li>python38Packages.m2r</li>
    <li>python38Packages.marshmallow</li>
    <li>python38Packages.mistune (python38Packages.mistune_0_8)</li>
    <li>python38Packages.mongoengine</li>
    <li>python38Packages.mpmath</li>
    <li>python38Packages.munkres</li>
    <li>python38Packages.nest-asyncio</li>
    <li>python38Packages.nose</li>
    <li>python38Packages.nose-exclude</li>
    <li>python38Packages.nose_warnings_filters</li>
    <li>python38Packages.olefile</li>
    <li>python38Packages.openpyxl</li>
    <li>python38Packages.ordereddict</li>
    <li>python38Packages.outcome</li>
    <li>python38Packages.pandocfilters</li>
    <li>python38Packages.pathpy</li>
    <li>python38Packages.pexpect</li>
    <li>python38Packages.pickleshare</li>
    <li>python38Packages.ply</li>
    <li>python38Packages.pretend</li>
    <li>python38Packages.prometheus_client</li>
    <li>python38Packages.purl</li>
    <li>python38Packages.pyext</li>
    <li>python38Packages.pyhamcrest</li>
    <li>python38Packages.pylama</li>
    <li>python38Packages.pymongo</li>
    <li>python38Packages.pyperf</li>
    <li>python38Packages.pyroma</li>
    <li>python38Packages.pysendfile</li>
    <li>python38Packages.pytest-asyncio</li>
    <li>python38Packages.pytest-flake8</li>
    <li>python38Packages.pytest-localserver</li>
    <li>python38Packages.pytest-timeout</li>
    <li>python38Packages.pythonImportsCheckHook</li>
    <li>python38Packages.python_mimeparse</li>
    <li>python38Packages.pyudev</li>
    <li>python38Packages.recursivePthLoader</li>
    <li>python38Packages.redis</li>
    <li>python38Packages.rednose</li>
    <li>python38Packages.responses</li>
    <li>python38Packages.rsa</li>
    <li>python38Packages.send2trash</li>
    <li>python38Packages.sortedcontainers</li>
    <li>python38Packages.soupsieve</li>
    <li>python38Packages.terminado</li>
    <li>python38Packages.termstyle</li>
    <li>python38Packages.testresources</li>
    <li>python38Packages.testscenarios</li>
    <li>python38Packages.testtools</li>
    <li>python38Packages.text-unidecode</li>
    <li>python38Packages.tornado</li>
    <li>python38Packages.traitlets</li>
    <li>python38Packages.ukpostcodeparser</li>
    <li>python38Packages.vcversioner</li>
    <li>python38Packages.voluptuous</li>
    <li>python38Packages.werkzeug</li>
    <li>python38Packages.wrapPython</li>
    <li>python38Packages.xlrd</li>
    <li>python38Packages.xlwt</li>
    <li>python38Packages.xmltodict</li>
    <li>python38Packages.zope_event</li>
    <li>python38Packages.zope_exceptions</li>
    <li>python38Packages.zope_interface</li>
    <li>python38Packages.zope_location</li>
    <li>python38Packages.zope_proxy</li>
    <li>python38Packages.zope_testing</li>
    <li>python39 (python39Packages.python)</li>
    <li>python39Packages.alabaster</li>
    <li>python39Packages.appdirs</li>
    <li>python39Packages.async-timeout</li>
    <li>python39Packages.atomicwrites</li>
    <li>python39Packages.attrs</li>
    <li>python39Packages.automat</li>
    <li>python39Packages.azure-common</li>
    <li>python39Packages.azure-nspkg</li>
    <li>python39Packages.bootstrapped-pip</li>
    <li>python39Packages.certifi</li>
    <li>python39Packages.colorama</li>
    <li>python39Packages.constantly</li>
    <li>python39Packages.cookies</li>
    <li>python39Packages.coverage</li>
    <li>python39Packages.dateutil (python39Packages.python-dateutil)</li>
    <li>python39Packages.decorator</li>
    <li>python39Packages.defusedxml</li>
    <li>python39Packages.distlib</li>
    <li>python39Packages.docutils</li>
    <li>python39Packages.ecdsa</li>
    <li>python39Packages.eradicate</li>
    <li>python39Packages.extras</li>
    <li>python39Packages.filelock</li>
    <li>python39Packages.fixtures</li>
    <li>python39Packages.greenlet</li>
    <li>python39Packages.imagesize</li>
    <li>python39Packages.importlib-metadata</li>
    <li>python39Packages.incremental</li>
    <li>python39Packages.iniconfig</li>
    <li>python39Packages.ipython_genutils</li>
    <li>python39Packages.isodate</li>
    <li>python39Packages.itsdangerous</li>
    <li>python39Packages.jupyterlab-pygments</li>
    <li>python39Packages.lazy-object-proxy</li>
    <li>python39Packages.linecache2</li>
    <li>python39Packages.m2r</li>
    <li>python39Packages.markupsafe</li>
    <li>python39Packages.mistune (python39Packages.mistune_0_8)</li>
    <li>python39Packages.mock</li>
    <li>python39Packages.mongoengine</li>
    <li>python39Packages.more-itertools</li>
    <li>python39Packages.nest-asyncio</li>
    <li>python39Packages.nose</li>
    <li>python39Packages.nose-exclude</li>
    <li>python39Packages.nose_warnings_filters</li>
    <li>python39Packages.olefile</li>
    <li>python39Packages.ordereddict</li>
    <li>python39Packages.outcome</li>
    <li>python39Packages.packaging</li>
    <li>python39Packages.pandocfilters</li>
    <li>python39Packages.pbr</li>
    <li>python39Packages.pexpect</li>
    <li>python39Packages.pip</li>
    <li>python39Packages.pipBuildHook</li>
    <li>python39Packages.pipInstallHook</li>
    <li>python39Packages.pkgconfig</li>
    <li>python39Packages.pluggy</li>
    <li>python39Packages.ply</li>
    <li>python39Packages.pretend</li>
    <li>python39Packages.prometheus_client</li>
    <li>python39Packages.psutil</li>
    <li>python39Packages.ptyprocess</li>
    <li>python39Packages.purl</li>
    <li>python39Packages.py</li>
    <li>python39Packages.pyasn1</li>
    <li>python39Packages.pycodestyle</li>
    <li>python39Packages.pycparser</li>
    <li>python39Packages.pyext</li>
    <li>python39Packages.pyflakes</li>
    <li>python39Packages.pygments</li>
    <li>python39Packages.pyhamcrest</li>
    <li>python39Packages.pymongo</li>
    <li>python39Packages.pyparsing</li>
    <li>python39Packages.pyperf</li>
    <li>python39Packages.pysocks</li>
    <li>python39Packages.pythonCatchConflictsHook</li>
    <li>python39Packages.pythonImportsCheckHook</li>
    <li>python39Packages.pythonNamespacesHook</li>
    <li>python39Packages.pythonRemoveTestsDirHook</li>
    <li>python39Packages.python_mimeparse</li>
    <li>python39Packages.pytz</li>
    <li>python39Packages.redis</li>
    <li>python39Packages.rednose</li>
    <li>python39Packages.ruamel-base (python39Packages.ruamel_base)</li>
    <li>python39Packages.semantic-version</li>
    <li>python39Packages.setuptools</li>
    <li>python39Packages.setuptools-rust</li>
    <li>python39Packages.setuptools-scm (python39Packages.setuptools_scm)</li>
    <li>python39Packages.setuptoolsBuildHook</li>
    <li>python39Packages.setuptoolsCheckHook</li>
    <li>python39Packages.six</li>
    <li>python39Packages.sortedcontainers</li>
    <li>python39Packages.soupsieve</li>
    <li>python39Packages.sphinxcontrib-applehelp</li>
    <li>python39Packages.sphinxcontrib-devhelp</li>
    <li>python39Packages.sphinxcontrib-htmlhelp</li>
    <li>python39Packages.sphinxcontrib-jsmath</li>
    <li>python39Packages.sphinxcontrib-qthelp</li>
    <li>python39Packages.sphinxcontrib-serializinghtml</li>
    <li>python39Packages.sphinxcontrib-websupport</li>
    <li>python39Packages.terminado</li>
    <li>python39Packages.termstyle</li>
    <li>python39Packages.testpath</li>
    <li>python39Packages.testresources</li>
    <li>python39Packages.testscenarios</li>
    <li>python39Packages.testtools</li>
    <li>python39Packages.toml</li>
    <li>python39Packages.tomlkit</li>
    <li>python39Packages.tornado</li>
    <li>python39Packages.traceback2</li>
    <li>python39Packages.typing-extensions</li>
    <li>python39Packages.u-msgpack-python</li>
    <li>python39Packages.ukpostcodeparser</li>
    <li>python39Packages.unittest2</li>
    <li>python39Packages.vcversioner</li>
    <li>python39Packages.wcwidth</li>
    <li>python39Packages.wheel</li>
    <li>python39Packages.wheelUnpackHook</li>
    <li>python39Packages.wrapPython</li>
    <li>python39Packages.wrapt</li>
    <li>python39Packages.zipp</li>
    <li>python39Packages.zope_event</li>
    <li>python39Packages.zope_interface</li>
    <li>remind</li>
    <li>rset</li>
    <li>rtl8723bs-firmware</li>
    <li>rtmpdump</li>
    <li>sbclBootstrap</li>
    <li>scaleft</li>
    <li>scid</li>
    <li>serd</li>
    <li>shen-sources</li>
    <li>shhopt</li>
    <li>softmaker-office</li>
    <li>sord</li>
    <li>source-han-mono</li>
    <li>speedread</li>
    <li>ssh-ident</li>
    <li>subdl</li>
    <li>tcl (tcl-8_6)</li>
    <li>tcl-8_5</li>
    <li>tcl2048</li>
    <li>tcllib</li>
    <li>tclreadline</li>
    <li>tcltls</li>
    <li>tclx</li>
    <li>tix</li>
    <li>tk (tk-8_6)</li>
    <li>tk-8_5</li>
    <li>tmuxPlugins.net-speed</li>
    <li>tnat64</li>
    <li>tt-rss-theme-feedly</li>
    <li>umount (unixtools.umount)</li>
    <li>unicode-character-database</li>
    <li>unicode-emoji</li>
    <li>unixtools.netstat</li>
    <li>unixtools.route</li>
    <li>watch (unixtools.watch)</li>
    <li>unixtools.whereis</li>
    <li>xxd (unixtools.xxd)</li>
    <li>unzip</li>
    <li>updateAutotoolsGnuConfigScriptsHook</li>
    <li>validatePkgConfig</li>
    <li>vimPlugins.NeoSolarized</li>
    <li>vimPlugins.NrrwRgn</li>
    <li>vimPlugins.acp</li>
    <li>vimPlugins.agda-vim</li>
    <li>vimPlugins.argtextobj-vim</li>
    <li>vimPlugins.auto-git-diff</li>
    <li>vimPlugins.auto-session</li>
    <li>vimPlugins.barbar-nvim</li>
    <li>vimPlugins.ci_dark</li>
    <li>vimPlugins.coc-clangd</li>
    <li>vimPlugins.coc-json</li>
    <li>vimPlugins.coc-pairs</li>
    <li>vimPlugins.coc-solargraph</li>
    <li>vimPlugins.colorizer</li>
    <li>vimPlugins.deoplete-dictionary</li>
    <li>vimPlugins.deoplete-fish</li>
    <li>vimPlugins.deoplete-lsp</li>
    <li>vimPlugins.deoplete-nvim</li>
    <li>vimPlugins.dracula-vim</li>
    <li>vimPlugins.fastfold</li>
    <li>vimPlugins.fennel-vim</li>
    <li>vimPlugins.gitv</li>
    <li>vimPlugins.glow-nvim</li>
    <li>vimPlugins.hiPairs</li>
    <li>vimPlugins.hologram-nvim</li>
    <li>vimPlugins.iosvkem</li>
    <li>vimPlugins.kotlin-vim</li>
    <li>vimPlugins.last256</li>
    <li>vimPlugins.limelight-vim</li>
    <li>vimPlugins.molokai</li>
    <li>vimPlugins.ncm2</li>
    <li>vimPlugins.ncm2-path</li>
    <li>vimPlugins.ncm2-tmux</li>
    <li>vimPlugins.neco-syntax</li>
    <li>vimPlugins.nerdtree</li>
    <li>vimPlugins.nvim-cm-racer</li>
    <li>vimPlugins.nvim-jdtls</li>
    <li>vimPlugins.nvim-peekup</li>
    <li>vimPlugins.nvim-toggleterm-lua</li>
    <li>vimPlugins.nvim-treesitter-pyfold</li>
    <li>vimPlugins.nvim-yarp</li>
    <li>vimPlugins.open-browser-github-vim</li>
    <li>vimPlugins.ranger-vim</li>
    <li>vimPlugins.stan-vim</li>
    <li>vimPlugins.tagbar</li>
    <li>vimPlugins.taglist-vim</li>
    <li>vimPlugins.tlib_vim</li>
    <li>vimPlugins.todo-txt-vim</li>
    <li>vimPlugins.utl-vim</li>
    <li>vimPlugins.vim-addon-actions</li>
    <li>vimPlugins.vim-addon-async</li>
    <li>vimPlugins.vim-addon-commenting</li>
    <li>vimPlugins.vim-addon-local-vimrc</li>
    <li>vimPlugins.vim-addon-manager</li>
    <li>vimPlugins.vim-addon-mw-utils</li>
    <li>vimPlugins.vim-addon-signs</li>
    <li>vimPlugins.vim-addon-toggle-buffer</li>
    <li>vimPlugins.vim-android</li>
    <li>vimPlugins.vim-beancount</li>
    <li>vimPlugins.vim-bufkill</li>
    <li>vimPlugins.vim-closer</li>
    <li>vimPlugins.vim-concourse</li>
    <li>vimPlugins.vim-csharp</li>
    <li>vimPlugins.vim-docbk</li>
    <li>vimPlugins.vim-easytags</li>
    <li>vimPlugins.vim-endwise</li>
    <li>vimPlugins.vim-fireplace</li>
    <li>vimPlugins.vim-fish</li>
    <li>vimPlugins.vim-gui-position</li>
    <li>vimPlugins.vim-hier</li>
    <li>vimPlugins.vim-indent-guides</li>
    <li>vimPlugins.vim-janah</li>
    <li>vimPlugins.vim-lastplace</li>
    <li>vimPlugins.vim-lawrencium</li>
    <li>vimPlugins.vim-mergetool</li>
    <li>vimPlugins.vim-merginal</li>
    <li>vimPlugins.vim-misc</li>
    <li>vimPlugins.vim-nerdtree-syntax-highlight</li>
    <li>vimPlugins.vim-obsession</li>
    <li>vimPlugins.vim-ocaml</li>
    <li>vimPlugins.vim-osc52</li>
    <li>vimPlugins.vim-qf</li>
    <li>vimPlugins.vim-racket</li>
    <li>vimPlugins.vim-ragtag</li>
    <li>vimPlugins.vim-shellcheck</li>
    <li>vimPlugins.vim-tpipeline</li>
    <li>vimPlugins.vim-unimpaired</li>
    <li>vimPlugins.vim-vsnip-integ</li>
    <li>vimer</li>
    <li>volk</li>
    <li>vscode-extensions.editorconfig.editorconfig</li>
    <li>vscode-extensions.stephlin.vscode-tmux-keybinding</li>
    <li>vscode-extensions.usernamehw.errorlens</li>
    <li>vscode-extensions.yzhang.markdown-all-in-one</li>
    <li>vulkan-headers</li>
    <li>wayland-protocols</li>
    <li>wayland-scanner</li>
    <li>webos.cmake-modules</li>
    <li>xfce.automakeAddFlags</li>
    <li>xhtml1</li>
    <li>xkeyboard_config (xorg.xkeyboardconfig)</li>
    <li>xorg.libICE</li>
    <li>xorg.libSM</li>
    <li>xorg.libX11</li>
    <li>xorg.libXau</li>
    <li>xorg.libXcomposite</li>
    <li>xorg.libXcursor</li>
    <li>xorg.libXdamage</li>
    <li>xorg.libXdmcp</li>
    <li>xorg.libXext</li>
    <li>xorg.libXfixes</li>
    <li>xorg.libXft</li>
    <li>xorg.libXi</li>
    <li>xorg.libXinerama</li>
    <li>xorg.libXmu</li>
    <li>xorg.libXpm</li>
    <li>xorg.libXrandr</li>
    <li>xorg.libXrender</li>
    <li>xorg.libXres</li>
    <li>xorg.libXt</li>
    <li>xorg.libXtst</li>
    <li>xorg.libXv</li>
    <li>xorg.libXvMC</li>
    <li>xorg.libXxf86vm</li>
    <li>xorg.libfontenc</li>
    <li>xorg.libpciaccess</li>
    <li>xorg.libpthreadstubs</li>
    <li>xorg.libxcb</li>
    <li>xorg.lndir</li>
    <li>xorg.utilmacros</li>
    <li>xorg.xcbproto</li>
    <li>xorg.xcbutil</li>
    <li>xorg.xcbutilimage</li>
    <li>xorg.xcbutilkeysyms</li>
    <li>xorg.xcbutilrenderutil</li>
    <li>xorg.xcbutilwm</li>
    <li>xorg.xorgproto</li>
    <li>xorg.xtrans</li>
    <li>yarn</li>
    <li>zenstates</li>
    <li>zinit</li>
    <li>zsh-autopair</li>
    <li>zsh-powerlevel9k</li>
    <li>zsh-you-should-use</li>
    <li>zxing-cpp</li>
  </ul>
</details>

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

primeos committed:
Merge pull request #119558 from primeos/mesa-next

mesa: 21.0.3 -> 21.1.1

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

mweinelt committed:
Revert "python3Packages.pywemo: disable failing test"

This reverts commit a0160c0ae804fa12a98f7020e786fc2565b81974.

The original issue was a result of a libxml2 regression that has since
been fixed.

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

mweinelt committed:
python3Packages.SQLAlchemy: 1.4.15 -> 1.4.16

https://github.com/sqlalchemy/sqlalchemy/releases/tag/rel_1_4_16

mweinelt committed:
python3Packages.SQLAlchemy: don't run tests in parallel

Test failures on hydra look suspiciously like they stem from xdist.

gw0 ok / gw1 ok / gw2 ok / gw3 ok / gw4 ok / gw5 ok / gw6 ok / gw7 ok / gw8 ok / gw9 ok / gw10 ok / gw11 ok / gw12 ok / gw13 ok / gw14 ok / gw15 ok / gw16 ok / gw17 ok / gw18 ok / gw19 C / gw20 C / gw21 C / gw22 C / gw23 C / gw24 C / gw25 C / gw26 C / gw27 C / gw28 C / gw29 C / gw30 C / gw31 C / gw32 C / gw33 C / gw34 C / gw35 C / gw36 C / gw37 C / gw38 C / gw39 C / gw40 C / gw41 C / gw42 C / gw43 C / gw44 C / gw45 C / gw46 C / gw47 CINTERNALERROR> def worker_internal_error(self, node, formatted_error):
INTERNALERROR>         """
INTERNALERROR>         pytest_internalerror() was called on the worker.
INTERNALERROR>
INTERNALERROR>         pytest_internalerror() arguments are an excinfo and an excrepr, which can't
INTERNALERROR>         be serialized, so we go with a poor man's solution of raising an exception
INTERNALERROR>         here ourselves using the formatted message.
INTERNALERROR>         """
INTERNALERROR>         self._active_nodes.remove(node)
INTERNALERROR>         try:
INTERNALERROR> >           assert False, formatted_error
INTERNALERROR> E           AssertionError: Traceback (most recent call last):
INTERNALERROR> E               File "/nix/store/4v6lhz8sq9jwl3af8abs6cgpbh9lignr-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/main.py", line 267, in wrap_session
INTERNALERROR> E                 config.hook.pytest_sessionstart(session=session)
INTERNALERROR> E               File "/nix/store/r4lwmmknxwx3gq2bv73yf0rkli9d902d-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR> E                 return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR> E               File "/nix/store/r4lwmmknxwx3gq2bv73yf0rkli9d902d-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR> E                 return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR> E               File "/nix/store/r4lwmmknxwx3gq2bv73yf0rkli9d902d-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
INTERNALERROR> E                 self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR> E               File "/nix/store/r4lwmmknxwx3gq2bv73yf0rkli9d902d-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR> E                 return outcome.get_result()
INTERNALERROR> E               File "/nix/store/r4lwmmknxwx3gq2bv73yf0rkli9d902d-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR> E                 raise ex[1].with_traceback(ex[2])
INTERNALERROR> E               File "/nix/store/r4lwmmknxwx3gq2bv73yf0rkli9d902d-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR> E                 res = hook_impl.function(*args)
INTERNALERROR> E               File "/build/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/plugin/pytestplugin.py", line 135, in pytest_sessionstart
INTERNALERROR> E                 asyncio._assume_async(plugin_base.post_begin)
INTERNALERROR> E               File "/nix/store/03hwa29jf7794x48983j44g0qvancijw-python3.8-SQLAlchemy-1.4.15/lib/python3.8/site-packages/sqlalchemy/testing/asyncio.py", line 50, in _assume_async
INTERNALERROR> E                 return _util_async_run(fn, *args, **kwargs)
INTERNALERROR> E               File "/nix/store/03hwa29jf7794x48983j44g0qvancijw-python3.8-SQLAlchemy-1.4.15/lib/python3.8/site-packages/sqlalchemy/util/_concurrency_py3k.py", line 167, in _util_async_run
INTERNALERROR> E                 loop = asyncio.get_event_loop()
INTERNALERROR> E               File "/nix/store/4s0h5aawbap3xhldxhcijvl26751qrjr-python3-3.8.9/lib/python3.8/asyncio/events.py", line 639, in get_event_loop
INTERNALERROR> E                 raise RuntimeError('There is no current event loop in thread %r.'
INTERNALERROR> E             RuntimeError: There is no current event loop in thread 'Dummy-1'.
INTERNALERROR> E           assert False
INTERNALERROR>
INTERNALERROR> /nix/store/ws9fk09ssyzhy93i8janlh274nay3190-python3.8-pytest-xdist-2.2.1/lib/python3.8/site-packages/xdist/dsession.py:187: AssertionError
[gw18] node down: Not properly terminated

replacing crashed worker gw18
[gw48] linux Python 3.8.9 cwd: /build/SQLAlchemy-1.4.15
gw0 ok / gw1 ok / gw2 ok / gw3 ok / gw4 ok / gw5 ok / gw6 ok / gw7 ok / gw8 ok / gw9 ok / gw10 ok / gw11 ok / gw12 ok / gw13 ok / gw14 ok / gw15 ok / gw16 ok / gw17 ok / gw48 C / gw19 C / gw20 C / gw21 C / gw22 C / gw23 C / gw24 C / gw25 C / gw26 C / gw27 C / gw28 C / gw29 C / gw30 C / gw31 C / gw32 C / gw33 C / gw34 C / gw35 C / gw36 C / gw37 C / gw38 C / gw39 C / gw40 C / gw41 C / gw42 C / gw43 C / gw44 C / gw45 C / gw46 C / gw47 CINTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/nix/store/4v6lhz8sq9jwl3af8abs6cgpbh9lignr-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/main.py", line 269, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/nix/store/4v6lhz8sq9jwl3af8abs6cgpbh9lignr-python3.8-pytest-6.2.3/lib/python3.8/site-packages/_pytest/main.py", line 323, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/nix/store/r4lwmmknxwx3gq2bv73yf0rkli9d902d-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/nix/store/r4lwmmknxwx3gq2bv73yf0rkli9d902d-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/nix/store/r4lwmmknxwx3gq2bv73yf0rkli9d902d-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
INTERNALERROR>     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR>   File "/nix/store/r4lwmmknxwx3gq2bv73yf0rkli9d902d-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/nix/store/r4lwmmknxwx3gq2bv73yf0rkli9d902d-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/nix/store/r4lwmmknxwx3gq2bv73yf0rkli9d902d-python3.8-pluggy-0.13.1/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/nix/store/ws9fk09ssyzhy93i8janlh274nay3190-python3.8-pytest-xdist-2.2.1/lib/python3.8/site-packages/xdist/dsession.py", line 112, in pytest_runtestloop
INTERNALERROR>     self.loop_once()
INTERNALERROR>   File "/nix/store/ws9fk09ssyzhy93i8janlh274nay3190-python3.8-pytest-xdist-2.2.1/lib/python3.8/site-packages/xdist/dsession.py", line 135, in loop_once
INTERNALERROR>     call(**kwargs)
INTERNALERROR>   File "/nix/store/ws9fk09ssyzhy93i8janlh274nay3190-python3.8-pytest-xdist-2.2.1/lib/python3.8/site-packages/xdist/dsession.py", line 224, in worker_errordown
INTERNALERROR>     self._active_nodes.remove(node)
INTERNALERROR> KeyError: <WorkerController gw18>

mweinelt commented:
I was just updating sqlalchemy when I noticed it was also blocking the manual in here. I rebased the commit on top of staging-next and disabled xdist, which seems to be the issue.

Upstream also uses xdist, but I bet our situation is a bit difference with regards to parallelism and congestion.

https://github.com/NixOS/nixpkgs/pull/124803

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

primeos committed:
mesa: 21.1.1 -> 21.1.2

github-actions[bot] committed:
Merge master into staging-next

mweinelt committed:
Merge branch 'master' into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

mweinelt commented:
@NixOS/darwin-maintainers mesa fails to build because it can't find libunwind, blocking Inkscape.

https://hydra.nixos.org/build/144152054

github-actions[bot] committed:
Merge master into staging-next

primeos committed:
Merge pull request #125424 from primeos/mesa

mesa: 21.1.1 -> 21.1.2

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

mweinelt commented:
@NixOS/darwin-maintainers The tarball jobs chokes on `darwin-arm64` attribute..

> attribute 'darwin-arm64' missing, at /build/source/pkgs/tools/networking/ngrok-2/default.nix:14:19

https://hydra.nixos.org/build/144759321

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

jonringer committed:
apple_sdk_11: alias JavaVM

Apple removed JavaVM but JavaNativeFoundation seems to be
the replacement. Should be safe to substitute usage of
JavaVM with JavaNativeFoundation.

jonringer committed:
mesa: fix darwin build

github-actions[bot] committed:
Merge master into staging-next

jonringer commented:
tarball was fixed, now mesa darwin regressions are blocking https://hydra.nixos.org/job/nixpkgs/staging-next/unstable#tabs-constituents

andersk commented:
Possibly a side effect of https://gitlab.freedesktop.org/mesa/mesa/-/commit/766538f83cbbe33ca291feb7ec43cf2937c098da “darwin: Use the system libunwind”?

jonringer commented:
I forgot to link it: #126021

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

mweinelt committed:
python3Packages.authcaptureproxy: relax importlib-metadata dependency

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

mweinelt committed:
Merge branch 'master' into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

mweinelt commented:
@NixOS/darwin-maintainers x86_64-darwin stdenv breakage in gnum4. https://hydra.nixos.org/build/145132415

```
In file included from /nix/store/zxyiijj7wrrzzc6fy18k3rj0pz353s1s-bootstrap-stage0-Libsystem/include/stddef.h:69:
/nix/store/zxyiijj7wrrzzc6fy18k3rj0pz353s1s-bootstrap-stage0-Libsystem/include/sys/_types/_ptrdiff_t.h:31:28: error: 'long type-name' is invalid
typedef __darwin_ptrdiff_t ptrdiff_t;
                           ^
./config.h:1749:19: note: expanded from macro 'ptrdiff_t'
#define ptrdiff_t long
                  ^
1 error generated.
make[3]: *** [Makefile:1915: gl_avltree_oset.o] Error 1
```

jonringer commented:
related: https://github.com/NixOS/nixpkgs/issues/126617

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

jtojnar committed:
Merge branch 'master' into staging-next

Regenerated pkgs/servers/x11/xorg/default.nix to resolve the conflict.

jonringer commented:
This has been going on a month...  I will be merging once darwin rebuilds dip below 6k (currently 8.6k)

jonringer committed:
Merge branch 'master' into staging-next

Conflicts:
  pkgs/servers/x11/xorg/default.nix

jonringer committed:
xorg: fix bad merge conflict

github-actions[bot] committed:
Merge master into staging-next

vcunat commented:
Currently Hydra is scheduling darwin builds mainly for staging-20.09.

vcunat commented:
Overall it's sad when one platform significantly drags down others; hopefully that won't be happening so often.

Ashe commented:
Music to my ears, I've been so excited for this update! I need that mesa patch!

github-actions[bot] committed:
Merge master into staging-next

Synthetica9 commented:
r13y.com will thank you :)

Atemu commented:
Yeah, the minimal ISO should hopefully be 100% reproducible after staging. Could you trigger another check once this is merged @grahamc?

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge master into staging-next

jonringer commented:
Okay, the x86_64-darwin rebuilds haven't moved in 24 hours... I'm sorry darwin users, but at this rate staging-next would be permanently paused.
